src/EbmlSInteger.cpp

```diff
@@ -154,7 +138,7 @@ filepos_t EbmlSInteger::ReadData(IOCallback & input, ScopeMode ReadFully)
     TempValue |= Buffer.at(i);
   }
 
-  Value = ToSigned(TempValue);
+  Value = std::make_signed_t<uint64>(TempValue);
```

Comment:
[make_signed](https://en.cppreference.com/w/cpp/types/make_signed) looks like is just changing the type. ToSigned () evaluates the binary data to turn it into a proper signed value.

Reply:
Yeah, the `std::make_signed_t<>` thingy is just a type, it isn't a function. So what you've written here is basically the same as the old C-style cast of `Value = (int64)TempValue;`, and that's certainly not what we want as converting an unsigned integer to a signed one with values that cannot be expressed by the signed type is undefined behavior.